packages/truffle-config/index.js

```diff
@@ -370,6 +382,27 @@ Config.detect = (options = {}, filename) => {
   return Config.load(file, options);
 };
 
+// When new options are passed in, a new eventManager needs to be
+// attached as it might override some options (e.g. { quiet: true })
+const attachNewEventManager = (config, newOptions) => {
+  const currentEventManagerOptions = config.eventManager.initializationOptions;
+  const { quiet, logger, globalConfig } = newOptions;
+  const optionsToMerge = {};
+
```

Comment:
Can we do this differently? I would love to just pull out the values but in the case that the new values are undefined I don't want it to overwrite the old stuff that we want to keep. Let me know if you have any ideas.